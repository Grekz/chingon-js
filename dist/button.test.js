"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const enzyme_1 = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");
const enzyme_to_json_1 = require("enzyme-to-json");
const jest_emotion_1 = require("jest-emotion");
const React = require("react");
const _1 = require(".");
enzyme_1.configure({ adapter: new Adapter() });
expect.extend(jest_emotion_1.matchers);
expect.addSnapshotSerializer(jest_emotion_1.default);
describe("<Button /> rendering", () => {
    it("should render Button", () => {
        const wrapper = enzyme_1.shallow(React.createElement(_1.default, null));
        expect(wrapper).toHaveLength(1);
    });
    it("hidden modal matches the snapshot", () => {
        const tree = enzyme_1.shallow(React.createElement(_1.default, null));
        expect(enzyme_to_json_1.default(tree)).toMatchSnapshot();
    });
    it("shown modal matches the snapshot", () => {
        const tree = enzyme_1.shallow(React.createElement(_1.default, { color: "white" }));
        expect(enzyme_to_json_1.default(tree)).toMatchSnapshot();
    });
});
//# sourceMappingURL=button.test.js.map