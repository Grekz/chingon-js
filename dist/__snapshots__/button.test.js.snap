// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> rendering hidden modal matches the snapshot 1`] = `
<Styled(button)>
  <Component />
</Styled(button)>
`;

exports[`<Button /> rendering shown modal matches the snapshot 1`] = `
<Styled(button)>
  <Component />
</Styled(button)>
`;
